File: apps/backend/backend/src/main/kotlin/cz/obec/portal/ruian/domain/CzechPoint.kt
package cz.obec.portal.ruian.domain

/**
 * A Czech POINT assisted service point (Czech POINT / Czech POINT@office)
 * returned by the Czech POINT locator. `distanceMeters` and `walkingMinutes`
 * are populated by [cz.obec.portal.ruian.service.CzechPointLocatorService].
 */
data class CzechPoint(
    val id: String,
    val name: String,
    val address: String,
    val lat: Double,
    val lon: Double,
    val distanceMeters: Double? = null,
    val walkingMinutes: Int? = null,
    val openingHours: String? = null,
    val services: List<String> = emptyList(),
)

File: apps/mobile/lib/features/address/domain/czech_point.dart
/// A Czech POINT assisted service point as returned by the locator API.
class CzechPoint {
  const CzechPoint({
    required this.id,
    required this.name,
    required this.address,
    required this.lat,
    required this.lon,
    this.distanceMeters,
    this.walkingMinutes,
    this.openingHours,
    this.services = const [],
  });

  final String id;
  final String name;
  final String address;
  final double lat;
  final double lon;
  final double? distanceMeters;
  final int? walkingMinutes;
  final String? openingHours;
  final List<String> services;

  factory CzechPoint.fromJson(Map<String, dynamic> json) {
    return CzechPoint(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      address: json['address'] as String? ?? '',
      lat: (json['lat'] as num?)?.toDouble() ?? 0,
      lon: (json['lon'] as num?)?.toDouble() ?? 0,
      distanceMeters: (json['distanceMeters'] as num?)?.toDouble(),
      walkingMinutes: (json['walkingMinutes'] as num?)?.toInt(),
      openingHours: json['openingHours'] as String?,
      services: (json['services'] as List?)?.cast<String>() ?? const [],
    );
  }

  /// Human-readable distance, e.g. "850 m" or "1,2 km".
  String get distanceLabel {
    final meters = distanceMeters;
    if (meters == null) return '—';
    if (meters < 1000) return '${meters.round()} m';
    final km = (meters / 1000).toStringAsFixed(1).replaceAll('.', ',');
    return '$km km';
  }

  /// Human-readable walking time, e.g. "10 min".
  String get walkingLabel =>
      walkingMinutes == null ? '—' : '$walkingMinutes min';
}
